promql/engine.go

```diff
@@ -914,7 +914,7 @@ func (ev *evaluator) rangeEval(f func([]parser.Value, *EvalNodeHelper) (Vector,
 	}
 	enh := &EvalNodeHelper{out: make(Vector, 0, biggestLen)}
 	seriess := make(map[uint64]Series, biggestLen) // Output series by series hash.
-	tempNumSamples := ev.currentSamples
+	tempNumSamples := 0
```

Comment:
This doesn't look right, this is ignoring the existence of anything that is memory when this function is called.

Reply:
Make sense. It's not right.

I went through code again and find that there are three step for func `rangeEval`

```
- Step 1: func evalExpr (e []Expr) => (m []Matrix)
- Step 2: func matrixColumn (m []Matrix) => (v []Vector)  // temporary step
- Step 3: func eval(v []Vector, nh NodeHelp) => (result []Vector)
```

Since `Step 2` is counted, currentSamples is weird but exact.

Reply:
Maybe more efficient implements for `rangeEval` here, however, it's a minor optimization.

Reply:
What are you proposing?

Reply:
Intuitively, the involved samples a n*1 metric vector multiply a constant should be `n+1+n`, while prometheus return `n+1n+1+n` due to implementation.

That is not easy to understand for fresher of prometheus.

Reply:
I'm not following, what does that have to do with this code?

Reply:
I mean involved samples is not correct intuitively.

But this PR is not a proper fix. I will close it.